### PR TITLE
Add refresh command to view title

### PR DESCRIPTION
**Pull request by ckaczor**
Attempted fix for #200 

Couldn't find any way to add buttons to the main title (vscode does it in places but couldn't find an extension that does it) - figuring this is the next best thing.

### CODE REVIEW

**Comment by alefragnani** [comment]
Hi @ckaczor ,

First of all, thanks for you PR and sorry for taking so long to answer. I was a bit busy in the last couple of weeks, with my vacation and the preparations for speaking at [Embarcadero Conference](https://www.embarcaderoconference.com.br).

I didn't find any API for a title button either, so I must admit it only works for each "panel". About that, since you added a Refresh button for each panel, the `RefreshProjects` command should refresh only the referred kind of project, instead of all of them. 

I'm not sure, but I think you can't pass parameters to that commands (on `view/title`), so you probably have to create a new command for each kind of project. 

Thank you.

**Comment by ckaczor** [comment]
Thanks for the review and comments - I updated the commit with changes that should work.

File: src/extension.ts [hunk 1]
```diff
@@ -60,6 +60,11 @@ export function activate(context: vscode.ExtensionContext) {
     // register commands (here, because it needs to be used right below if an invalid JSON is present)
     vscode.commands.registerCommand("projectManager.saveProject", () => saveProject());
     vscode.commands.registerCommand("projectManager.refreshProjects", () => refreshProjects(true, true));
+    vscode.commands.registerCommand("projectManager.refreshVSCodeProjects", () => refreshProjectsByType("VSCode", vscLocator, projectProviderVSCode, true, true));
+    vscode.commands.registerCommand("projectManager.refreshGitProjects", () => refreshProjectsByType("Git", gitLocator, projectProviderGit, true, true));
+    vscode.commands.registerCommand("projectManager.refreshSVNProjects", () => refreshProjectsByType("Mercurial", mercurialLocator, projectProviderMercurial, true, true));
```
Comment (alefragnani) [inline src/extension.ts]: The Refresh for SVN is connected to Mercurial 😄

File: src/extension.ts [hunk 2]
```diff
@@ -60,6 +60,11 @@ export function activate(context: vscode.ExtensionContext) {
...
+    vscode.commands.registerCommand("projectManager.refreshAnyProjects", () => refreshProjectsByType("SVN", svnLocator, projectProviderSVN, true, true));
```
Comment (alefragnani) [inline src/extension.ts]: The Refresh for ANY is connected to SVN 😄

File: src/extension.ts [hunk 3]
```diff
@@ -60,6 +60,11 @@ export function activate(context: vscode.ExtensionContext) {
...
+    vscode.commands.registerCommand("projectManager.refreshMercurialProjects", () => refreshProjectsByType("Any", anyLocator, projectProviderAny, true, true));
```
Comment (alefragnani) [inline src/extension.ts]: The Refresh for HG is connected to ANY 😄

File: src/extension.ts [hunk 4]
```diff
@@ -222,7 +227,32 @@ export function activate(context: vscode.ExtensionContext) {
...
+    }
+
+    async function delay(ms: number) {
+        return new Promise(resolve => setTimeout(resolve, ms));
+    }
 
+    function refreshProjectsByType(projectType: string, locator: CustomProjectLocator, projectProvider: ProjectProvider, showMessage?: boolean, forceRefresh?: boolean) {
+        vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: `Refreshing ${projectType} Projects`,
```
Comment (alefragnani) [inline src/extension.ts]: The progress message has changed. Now it shows: "`Refreshing Projects: ${projectType}"`, so you should remove `${projectType}` from this message and add it in the next `progress.report({...` line.

**Comment by alefragnani** [comment]
Hi @ckaczor,

I have noted a few changes to be made. Please, take a look and update the PR

Thank you 👍

**Comment by ckaczor** [comment]
Not enough coffee for me on a few of those copy/paste errors.  Fixes should be in - had some trouble squashing after merging in the upstream changes but I think it looks good.

Thanks!

**Pull request merged**